pkg/catalogv2/content/content.go

```diff
@@ -162,6 +156,148 @@ func (c *Manager) Index(namespace, name string, skipFilter bool) (*repo.IndexFil
 	return c.filterReleases(deepCopyIndex(index), k8sVersion, skipFilter), nil
 }
 
+// Icon Returns an io.ReadCloser and the icon's MIME type for the chart.
+//
+// If the chart's icon is not an HTTP or HTTPS URL, retrieves the icon from the repo's Git repository.
+// Otherwise, retrieves the icon via HTTP from the chart's URL and returns it as an io.ReadCloser with the proper Secret.
+func (c *Manager) Icon(namespace, name, chartName, version string) (io.ReadCloser, string, error) {
+	index, err := c.Index(namespace, name, true)
+	if err != nil {
+		return nil, "", err
+	}
+
+	chart, err := index.Get(chartName, version)
+	if err != nil {
+		return nil, "", err
+	}
+
+	repo, err := c.getRepo(namespace, name)
+	if err != nil {
+		return nil, "", err
+	}
+
+	if !isHTTP(chart.Icon) && repo.status.Commit != "" {
+		return git.Icon(namespace, name, repo.status.URL, chart)
+	}
+
+	secret, err := catalogv2.GetSecret(c.secrets, repo.spec, repo.metadata.Namespace)
+	if err != nil {
+		return nil, "", err
+	}
+
+	return helmhttp.Icon(secret, repo.status.URL, repo.spec.CABundle, repo.spec.InsecureSkipTLSverify, repo.spec.DisableSameOriginCheck, chart)
+}
+
+// Chart retrieves a specific Helm chart from a Helm repository.
+//
+// Retrieves the index file, fetches the helm chart and repository data.
+// Check's the commit status of the repository
+//
+// If the commit status of the repository is not an empty string,
+// Return the Chart through Git without secret
+//
+// If the commit status of the repository is an empty string,
+// it retrieves the secret associated with the repository
+//
+// The function returns an io.ReadCloser which represents the chart content.
+func (c *Manager) Chart(namespace, name, chartName, version string, skipFilter bool) (io.ReadCloser, error) {
+	// Get the index file of the Helm repo
+	index, err := c.Index(namespace, name, skipFilter)
+	if err != nil {
+		return nil, err
+	}
+
+	// Fetch specific Helm chart
+	chart, err := index.Get(chartName, version)
+	if err != nil {
+		return nil, err
+	}
+
+	// Retrieve the repository information
+	repo, err := c.getRepo(namespace, name)
+	if err != nil {
+		return nil, err
+	}
+
+	// If the commit status of the repository is not an empty string
+	// Return the Chart through Git without secret
```

Comment:
I didn't understand this. The secret is only used for the http connection to authenticate, we don't return the secret itself anywhere

Reply:
Yeah me neither, but I am changing the comment as to returning Git without checking the secret